addOns/frontendscanner/src/main/java/org/zaproxy/zap/extension/frontendscanner/FrontEndScannerProxyListener.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2020 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.frontendscanner;
+
+import java.util.List;
+import net.htmlparser.jericho.Element;
+import net.htmlparser.jericho.OutputDocument;
+import net.htmlparser.jericho.Source;
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.core.proxy.ProxyListener;
+import org.parosproxy.paros.extension.history.ProxyListenerLog;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.extension.api.API;
+
+/** The {@link ProxyListener} the {@link ExtensionFrontEndScanner} relies on. */
+public class FrontEndScannerProxyListener implements ProxyListener {
+    private static final Logger LOGGER = Logger.getLogger(FrontEndScannerProxyListener.class);
+
+    private FrontEndScannerAPI api;
+    private FrontEndScannerOptions options;
```

Comment:
`final`

Reply:
Should it be `final` in `ExtensionFrontEndScanner.java` too?
What about `api` here?
And what about `api`, and `proxyListener` in `ExtensionFrontEndScanner.java`?

Reply:
The comment is for both lines.
In the extension they can't be as they are not being initialised in the constructor.